src/frontend/src/optimizer/plan_node/logical_expand.rs

```diff
@@ -0,0 +1,252 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use itertools::Itertools;
+use risingwave_common::catalog::{Field, Schema};
+use risingwave_common::types::DataType;
+
+use super::{
+    BatchExpand, ColPrunable, LogicalProject, PlanBase, PlanRef, PlanTreeNodeUnary,
+    PredicatePushdown, StreamExpand, ToBatch, ToStream,
+};
+use crate::expr::InputRef;
+use crate::risingwave_common::error::Result;
+use crate::utils::{ColIndexMapping, Condition};
+
+/// [`LogicalExpand`] expand one row multiple times according to `column_subsets`.
+///
+/// It can be used to implement distinct aggregation and group set.
+#[derive(Debug, Clone)]
+pub struct LogicalExpand {
+    pub base: PlanBase,
+    column_subsets: Vec<Vec<usize>>,
+    input: PlanRef,
+}
```

Comment:
would you please add more comments? on the struct and the field `column_subsets`.